### PR TITLE
Homework12

### PR DESCRIPTION
**Pull request by userhost1**
rest of 12 task, marked like "to check"

### CODE REVIEW

**Comment by TheFallen058** [comment]
please try to implement everything in one commit) you made 13 commits for 10 files)

File: src/homework12/AdditionalTask_1.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+//  В исходном файле находятся слова, каждое слово на новой стороке.
+//  После запуска программы должен создать файл, который будет содержать в себе только палиндромы.
+
+public class AdditionalTask_1 {
+    public static void main(String[] args) throws FileNotFoundException {
```
Comment (TheFallen058) [inline src/homework12/AdditionalTask_1.java]: FileNotFoundException - no such exception is throwing

File: src/homework12/AdditionalTask_2.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+//        Определяем подходит ли нам предложение.
+//        Если подходит добавляем его в новый файл
+
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class AdditionalTask_2 {
+    public static void main(String[] args) {
+
+        StringBuffer text = new StringBuffer();
```
Comment (TheFallen058) [inline src/homework12/AdditionalTask_2.java]: no threads in this code so more efficient will be to use StringBuilder.
Name of the variable doesn't match with object it refers

File: src/homework12/AdditionalTask_3.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+public class AdditionalTask_3 {
+
+    private static int countSentences = 0;
+    private static ArrayList<String> sentences = new ArrayList<>();
+
+
+    public static void main(String[] args) {
+
+        StringBuffer text = new StringBuffer();
```
Comment (TheFallen058) [inline src/homework12/AdditionalTask_3.java]: same as in previous task

File: src/homework12/AdditionalTask_4.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            oos.writeObject(car1);
+            oos.writeObject(car2);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+
+        Car load1 = null;
```
Comment (TheFallen058) [inline src/homework12/AdditionalTask_4.java]: you don't need declaration of this variables here

File: src/homework12/Part3T34.java [hunk 5]
```diff
@@ -13,7 +13,7 @@
 public class Part3T34 {
     public static void main(String[] args) throws IOException {
```
Comment (TheFallen058) [inline src/homework12/Part3T34.java]: you don't throw such exception, redundant

File: src/homework12/Part3T35.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+package homework12;
+
+//  Записать в двоичный файл 20 случайных чисел.
+//  Прочитать записанный файл, распечатать числа и их среднее арифметическое.
+
+import java.io.*;
+
+public class Part3T35 {
+    public static void main(String[] args) throws IOException {
```
Comment (TheFallen058) [inline src/homework12/Part3T35.java]: same

**Review by TheFallen058 — CHANGES_REQUESTED** [review]
please check my comments

File: src/homework12/AdditionalTask_1.java (hunk 1 above)
Comment (userhost1) [reply src/homework12/AdditionalTask_1.java]: fixed on 
public static void main(String[] args) throws IOException {

File: src/homework12/AdditionalTask_2.java (hunk 2 above)
Comment (userhost1) [reply src/homework12/AdditionalTask_2.java]: changet on StringBuilder, but i dont understand what do u mean? -> Name of the variable doesn't match with object it refers

File: src/homework12/AdditionalTask_3.java (hunk 3 above)
Comment (userhost1) [reply src/homework12/AdditionalTask_3.java]: fixed

File: src/homework12/AdditionalTask_4.java (hunk 4 above)
Comment (userhost1) [reply src/homework12/AdditionalTask_4.java]: declaired in try catch method

File: src/homework12/AdditionalTask_2.java (hunk 2 above)
Comment (TheFallen058) [reply src/homework12/AdditionalTask_2.java]: https://www.javatpoint.com/difference-between-stringbuffer-and-stringbuilder#:~:text=The%20String%20class%20is%20an,and%20StringBuilder%20classes%20are%20mutable.&text=StringBuffer%20is%20synchronized%20i.e.%20thread,synchronized%20i.e.%20not%20thread%20safe.

**Review by TheFallen058 — APPROVED** [review]

**Pull request merged**